po-docs/ja/virt-tar.1
.\" Automatically generated by Podwrapper::Man 1.40.2 (Pod::Simple 3.35)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
.    ds C`
.    ds C'
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is >0, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.\"
.\" Avoid warning from groff about undefined register 'F'.
.de IX
..
.nr rF 0
.if \n(.g .if rF .nr rF 1
.if (\n(rF:(\n(.g==0)) \{\
.    if \nF \{\
.        de IX
.        tm Index:\\$1\t\\n%\t"\\$2"
..
.        if !\nF==2 \{\
.            nr % 0
.            nr F 2
.        \}
.    \}
.\}
.rr rF
.\" ========================================================================
.\"
.IX Title "virt-tar 1"
.TH virt-tar 1 "2019-02-07" "libguestfs-1.40.2" "Virtualization Support"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH "名前"
.IX Header "名前"
virt-tar \- 仮想マシンへのファイルの展開またはアップロード
.SH "書式"
.IX Header "書式"
.Vb 1
\& virt\-tar [\-\-options] \-x domname directory tarball
\&
\& virt\-tar [\-\-options] \-u domname tarball directory
\&
\& virt\-tar [\-\-options] disk.img [disk.img ...] \-x directory tarball
\&
\& virt\-tar [\-\-options] disk.img [disk.img ...] \-u tarball directory
.Ve
.SH "非推奨"
.IX Header "非推奨"
このツールは推奨されません。  代わりに \fBvirt\-copy\-in\fR\|(1), \fBvirt\-copy\-out\fR\|(1),
\&\fBvirt\-tar\-in\fR\|(1), \fBvirt\-tar\-out\fR\|(1) を使用してください。
.SH "例"
.IX Header "例"
仮想マシンから \f(CW\*(C`/home\*(C'\fR をローカルの tar ファイルにダウンロードします:
.PP
.Vb 1
\& virt\-tar \-x domname /home home.tar
\&
\& virt\-tar \-zx domname /home home.tar.gz
.Ve
.PP
ローカルの tar ファイルをアップロードして、仮想マシンの \f(CW\*(C`/tmp\*(C'\fR に展開します:
.PP
.Vb 1
\& virt\-tar \-u domname uploadstuff.tar /tmp
\&
\& virt\-tar \-zu domname uploadstuff.tar.gz /tmp
.Ve
.SH "警告"
.IX Header "警告"
You must \fInot\fR use \f(CW\*(C`virt\-tar\*(C'\fR with the \fI\-u\fR option (upload) on live
virtual machines.  If you do this, you risk disk corruption in the \s-1VM.\s0
\&\f(CW\*(C`virt\-tar\*(C'\fR tries to stop you from doing this, but doesn't catch all cases.
.PP
You can use \fI\-x\fR (extract) on live virtual machines, but you might get
inconsistent results or errors if there is filesystem activity inside the
\&\s-1VM.\s0  If the live \s-1VM\s0 is synched and quiescent, then \f(CW\*(C`virt\-tar\*(C'\fR will usually
work, but the only way to guarantee consistent results is if the virtual
machine is shut down.
.SH "説明"
.IX Header "説明"
\&\f(CW\*(C`virt\-tar\*(C'\fR is a general purpose archive tool for downloading and uploading
parts of a guest filesystem.  There are many possibilities: making backups,
uploading data files, snooping on guest activity, fixing or customizing
guests, etc.
.PP
If you want to just view a single file, use \fBvirt\-cat\fR\|(1).  If you just
want to edit a single file, use \fBvirt\-edit\fR\|(1).  For more complex cases you
should look at the \fBguestfish\fR\|(1) tool.
.PP
There are two modes of operation: \fI\-x\fR (eXtract) downloads a directory and
its contents (recursively) from the virtual machine into a local tarball.
\&\fI\-u\fR uploads from a local tarball, unpacking it into a directory inside the
virtual machine.  You cannot use these two options together.
.PP
In addition, you may need to use the \fI\-z\fR (gZip) option to enable
compression.  When uploading, you have to specify \fI\-z\fR if the upload file
is compressed because virt-tar won't detect this on its own.
.PP
\&\f(CW\*(C`virt\-tar\*(C'\fR can only handle tar (optionally gzipped) format tarballs.  For
example it cannot do PKZip files or bzip2 compression.  If you want that
then you'll have to rebuild the tarballs yourself.  (This is a limitation of
the \fBlibguestfs\fR\|(3) \s-1API\s0).
.SH "オプション"
.IX Header "オプション"
.IP "\fB\-\-help\fR" 4
.IX Item "--help"
簡単なヘルプを表示します。
.IP "\fB\-\-version\fR" 4
.IX Item "--version"
バージョン番号を表示して、終了します。
.IP "\fB\-c \s-1URI\s0\fR" 4
.IX Item "-c URI"
.PD 0
.IP "\fB\-\-connect \s-1URI\s0\fR" 4
.IX Item "--connect URI"
.PD
libvirt を使用していると、指定された \fI\s-1URI\s0\fR に接続します。  省略すると、デフォルトの libvirt ハイパーバイザーに接続します。
.Sp
仮想マシンのブロックデバイスを直接指定すると、libvirt はまったく使用されません。
.IP "\fB\-\-format\fR raw" 4
.IX Item "--format raw"
Specify the format of disk images given on the command line.  If this is
omitted then the format is autodetected from the content of the disk image.
.Sp
If disk images are requested from libvirt, then this program asks libvirt
for this information.  In this case, the value of the format parameter is
ignored.
.Sp
If working with untrusted raw-format guest disk images, you should ensure
the format is always specified.
.IP "\fB\-x\fR" 4
.IX Item "-x"
.PD 0
.IP "\fB\-\-extract\fR" 4
.IX Item "--extract"
.IP "\fB\-\-download\fR" 4
.IX Item "--download"
.IP "\fB\-u\fR" 4
.IX Item "-u"
.IP "\fB\-\-upload\fR" 4
.IX Item "--upload"
.PD
仮想マシンからローカル tar ファイルにディレクトリを展開 (ダウンロード) するには \fI\-x\fR を使用します。
.Sp
ローカル tar ファイルから仮想マシンの中にアップロードおよび展開するには \fI\-u\fR を使用します。このオプションを使用する前に \*(L"警告\*(R"
セクションを参照してください。
.Sp
これらのオプションのいずれかを正確に指定する必要があります。
.IP "\fB\-z\fR" 4
.IX Item "-z"
.PD 0
.IP "\fB\-\-gzip\fR" 4
.IX Item "--gzip"
.PD
入力または出力 tar ファイルを gzip 圧縮するよう指定します。
.SH "関連項目"
.IX Header "関連項目"
\&\fBguestfs\fR\|(3), \fBguestfish\fR\|(1), \fBvirt\-cat\fR\|(1), \fBvirt\-edit\fR\|(1),
\&\fBvirt\-copy\-in\fR\|(1), \fBvirt\-copy\-out\fR\|(1), \fBvirt\-tar\-in\fR\|(1),
\&\fBvirt\-tar\-out\fR\|(1), \fBSys::Guestfs\fR\|(3), \fBSys::Virt\fR\|(3),
http://libguestfs.org/.
.SH "著者"
.IX Header "著者"
Richard W.M. Jones http://people.redhat.com/~rjones/
.SH "COPYRIGHT"
.IX Header "COPYRIGHT"
Copyright (C) 2009 Red Hat Inc.
.SH "LICENSE"
.IX Header "LICENSE"
.SH "BUGS"
.IX Header "BUGS"
To get a list of bugs against libguestfs, use this link:
https://bugzilla.redhat.com/buglist.cgi?component=libguestfs&product=Virtualization+Tools
.PP
To report a new bug against libguestfs, use this link:
https://bugzilla.redhat.com/enter_bug.cgi?component=libguestfs&product=Virtualization+Tools
.PP
When reporting a bug, please supply:
.IP "\(bu" 4
The version of libguestfs.
.IP "\(bu" 4
Where you got libguestfs (eg. which Linux distro, compiled from source, etc)
.IP "\(bu" 4
Describe the bug accurately and give a way to reproduce it.
.IP "\(bu" 4
Run \fBlibguestfs\-test\-tool\fR\|(1) and paste the \fBcomplete, unedited\fR
output into the bug report.


ChangeLog
commit 91698a8500d19e06811133bfab27d1e1e65a583e
Author: Richard W.M. Jones <rjones@redhat.com>
Date:   Thu Feb 7 14:45:14 2019 +0000

    Version 1.40.2.

commit c7b374a757821bbd8f369fa268a75485163b52d7
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Wed Feb 6 15:19:53 2019 +0100

    inspect: revamp icon extraction for *SUSE guests
    
    Newer versions do not have the 24px distributor.png icon; OTOH they have
    the 48px version, so look for that one before the 24px one.
    
    Also, bump the size limit to 10K, as newer versions of the icon are
    bigger than 3K.
    
    (cherry picked from commit f41a75cea67dc3f2e18700303fac5d7078b3291a)

commit 8e62f30293737472ec6b9a2a238bb1206b2fbf4e
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Wed Feb 6 15:15:08 2019 +0100

    inspect: factor out find_png helper function
    
    Isolate the code for looking for a PNG file among a specified list from
    the icon_ubuntu implementation, and switch that function to it.
    
    This is just code refactoring, with no behaviour changes.
    
    (cherry picked from commit 6ff95acc72b02c4df0520e8cae7190447e4b2da5)

commit 178d9d821af6ee25f9f5141c20e4de42315de25f
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Wed Feb 6 15:05:51 2019 +0100

    inspect: bump size limit for Mageia guests
    
    Newer versions of the logo in Cauldron are almost 3K, so bump the limit
    to 10K for now.
    
    (cherry picked from commit 032b8d4c72ab233922482871403c2d9fe2f4a75b)

commit cc83aae48dbda40afa1b1652bd1a53dc9dd61573
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Tue Jan 29 15:07:18 2019 +0100

    v2v: linux: use NEVR for querying RPM packages (RHBZ#1669395)
    
    Use NEVR when querying RPM for the list of files of a package, instead
    of ENVR.  Also, use the epoch only when non-zero, and version of RPM
    supports it.
    
    The approach is basically copied from what supermin does in its RPM
    package handler.
    
    (cherry picked from commit 5eafd1273132b79018ddb72426b128f0fb8a39f4)

commit 76797064449a99351ca44586b16a50c5b74c0a47
Author: Richard W.M. Jones <rjones@redhat.com>
Date:   Mon Jan 21 15:59:45 2019 +0000

    build: Reduce -Wformat-overflow=2 to =1.
    
    GCC 9 gives this error:
    
    qemuopts.c: In function 'qemuopts_to_config_channel':
    qemuopts.c:987:29: error: '%.*s' directive output between 0 and 2147483647 bytes may exceed minimum required size of 4095 [-Werror=format-overflow=]
      987 |             fprintf (fp, "  %.*s = ", (int) k, values[j]);
          |                             ^~~~
    qemuopts.c:987:26: note: assuming directive output of 1 byte
      987 |             fprintf (fp, "  %.*s = ", (int) k, values[j]);
          |                          ^~~~~~~~~~~
    
    See discussion at the following GCC bug:
    
    https://gcc.gnu.org/bugzilla/show_bug.cgi?id=88993
    (cherry picked from commit b26bd778b58b78eb59503413fd5b41ac23725690)

commit d5dc8fba74edb39bede2e42316f3bd0e9c26c24c
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Tue Jan 22 13:04:59 2019 +0100

    python: change types for RBufferOut/FBuffer with Python 3 (RHBZ#1661871)
    
    So far RBufferOut return values, and FBuffer struct fields are 'str' on
    all the versions of Python.  Python 3 distinguishes between 'str'
    (unicode strings), and 'bytes', with 'str' no more able to hold
    arbitrary data.
    
    For this reason, switch the return value of RBufferOut functions, and
    FBuffer struct fields to bytes on Python 3: while this is a potentially
    incompatibile change, this is the only way to handle safely sequences
    of arbitrary bytes.
    
    (cherry picked from commit 0ee02e0117527b86a31b2a88a14994ce7f15571f)

commit c18957e0ea4233e2237e68c7d5e647cbde69e764
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Tue Jan 22 12:43:11 2019 +0100

    python: fix call of Python handlers of events
    
    Make sure to reference the arguments, to make sure they are kept alive
    during the function call; this is visible when setting an event handler
    for the CLOSE event, and testing it with Python 3.
    
    This does not seem to create a memory leak e.g. with Python 2.
    
    Also, switch away from the quasi-internal PyEval_CallObject to the
    public PyObject_CallObject, which takes care of doing safety checks.
    
    (cherry picked from commit 85235aec837716f1ddb2926b9a59a02543195500)

commit 53a7f2ffec1f7b014db4909371e0c1b1576ba10a
Author: Pino Toscano <ptoscano@redhat.com>
Date:   Fri Jan 18 13:22:26 2019 +0100

    build: integrate ocaml-link.sh with automake silent rules
    
    In case either the silent rules are disabled, or V=1 is set, print the
    full command line that is executed.
    
    (cherry picked from commit a8006afaf545d5188f592d8583f9dc373ded7671)

commit 1f026244ba5891a00617f4ab8c41b8680ff8c497
Author: Richard W.M. Jones <rjones@redhat.com>
Date:   Thu Jan 17 12:06:36 2019 +0000

    Version 1.40.1.

commit ae2aa14527605ffb89d86bf93bf8ca4f969118e1
Author: Richard W.M. Jones <rjones@redhat.com>
Date:   Thu Jan 17 12:03:07 2019 +0000

    valgrind: Add workaround for slow loading of debuginfo by valgrind.
    
    See nbdkit commit
    https://github.com/libguestfs/nbdkit/commit/bd32d899edb4c592a2df0ce4b26a2cf8396eba1d
    and https://bugzilla.redhat.com/show_bug.cgi?id=1662656
    
    (cherry picked from commit ea7c13fed7922b233db8382b1348dfb436e3ba85)

commit 1de72f8f00600bcaa76742dcfd366fb702eef3c2
Author: Richard W.M. Jones <rjones@redhat.com>
Date:   Tue Jan 15 14:39:41 2019 +0000

    Version 1.40.0.

commit 6151d8694d9db0d709a9ad807a5ca6a098d6502c
Author: Richard W.M. Jones <rjones@redhat.com>
Date:   Wed Jan 16 11:49:45 2019 +0000

    po: Drop ca and pt_BR from linguas.
    
    Commit 4252a490bb117940c1d38c838345ed110457386e dropped the PO files
    for ca (Catalan) and pt_BR (Brazilian Portuguese).  However it did not
    drop these languages from the linguas line in the Makefile resulting

po-docs/uk/virt-tar.pl
=encoding utf8

=head1 НАЗВА

virt-tar - Extract or upload files to a virtual machine

=head1 КОРОТКИЙ ОПИС

 virt-tar [--options] -x domname directory tarball

 virt-tar [--options] -u domname tarball directory

 virt-tar [--options] disk.img [disk.img ...] -x directory tarball

 virt-tar [--options] disk.img [disk.img ...] -u tarball directory

=head1 ЗАСТАРІЛЕ

This tool is obsolete.  Use L<virt-copy-in(1)>, L<virt-copy-out(1)>,
L<virt-tar-in(1)>, L<virt-tar-out(1)> as replacements.

=head1 ПРИКЛАДИ

Download C</home> from the VM into a local tarball:

 virt-tar -x domname /home home.tar

 virt-tar -zx domname /home home.tar.gz

Upload a local tarball and unpack it inside C</tmp> in the VM:

 virt-tar -u domname uploadstuff.tar /tmp

 virt-tar -zu domname uploadstuff.tar.gz /tmp

=head1 ПОПЕРЕДЖЕННЯ

You must I<not> use C<virt-tar> with the I<-u> option (upload) on live
virtual machines.  If you do this, you risk disk corruption in the VM.
C<virt-tar> tries to stop you from doing this, but doesn't catch all cases.

You can use I<-x> (extract) on live virtual machines, but you might get
inconsistent results or errors if there is filesystem activity inside the
VM.  If the live VM is synched and quiescent, then C<virt-tar> will usually
work, but the only way to guarantee consistent results is if the virtual
machine is shut down.

=head1 ОПИС

C<virt-tar> is a general purpose archive tool for downloading and uploading
parts of a guest filesystem.  There are many possibilities: making backups,
uploading data files, snooping on guest activity, fixing or customizing
guests, etc.

If you want to just view a single file, use L<virt-cat(1)>.  If you just
want to edit a single file, use L<virt-edit(1)>.  For more complex cases you
should look at the L<guestfish(1)> tool.

There are two modes of operation: I<-x> (eXtract) downloads a directory and
its contents (recursively) from the virtual machine into a local tarball.
I<-u> uploads from a local tarball, unpacking it into a directory inside the
virtual machine.  You cannot use these two options together.

In addition, you may need to use the I<-z> (gZip) option to enable
compression.  When uploading, you have to specify I<-z> if the upload file
is compressed because virt-tar won't detect this on its own.

C<virt-tar> can only handle tar (optionally gzipped) format tarballs.  For
example it cannot do PKZip files or bzip2 compression.  If you want that
then you'll have to rebuild the tarballs yourself.  (This is a limitation of
the L<libguestfs(3)> API).

=head1 ПАРАМЕТРИ

=over 4

=item B<--help>

Показати коротку довідку.

=item B<--version>

Показати дані щодо версії і завершити роботу.

=item B<-c адреса>

=item B<--connect адреса>

Якщо використовується libvirt, встановити з’єднання з вказаним I<URI>. Якщо
пропущено, з’єднання буде встановлено з типовим гіпервізором libvirt.

If you specify guest block devices directly, then libvirt is not used at
all.

=item B<--format> raw

Specify the format of disk images given on the command line.  If this is
omitted then the format is autodetected from the content of the disk image.

If disk images are requested from libvirt, then this program asks libvirt
for this information.  In this case, the value of the format parameter is
ignored.

If working with untrusted raw-format guest disk images, you should ensure
the format is always specified.

=item B<-x>

=item B<--extract>

=item B<--download>

=item B<-u>

=item B<--upload>

Use I<-x> to extract (download) a directory from a virtual machine to a
local tarball.

Use I<-u> to upload and unpack from a local tarball into a virtual machine.
Please read the L</WARNING> section above before using this option.

You must specify exactly one of these options.

=item B<-z>

=item B<--gzip>

Specify that the input or output tarball is gzip-compressed.

=back

=head1 ЕКРАНУВАННЯ СИМВОЛІВ У ОБОЛОНЦІ

Libvirt guest names can contain arbitrary characters, some of which have
meaning to the shell such as C<#> and space.  You may need to quote or
escape these characters on the command line.  See the shell manual page
L<sh(1)> for details.

=head1 ТАКОЖ ПЕРЕГЛЯНЬТЕ

L<guestfs(3)>, L<guestfish(1)>, L<virt-cat(1)>, L<virt-edit(1)>,
L<virt-copy-in(1)>, L<virt-copy-out(1)>, L<virt-tar-in(1)>,
L<virt-tar-out(1)>, L<Sys::Guestfs(3)>, L<Sys::Virt(3)>,
L<http://libguestfs.org/>.

=head1 АВТОР

Richard W.M. Jones L<http://people.redhat.com/~rjones/>

=head1 АВТОРСЬКІ ПРАВА

© Red Hat Inc., 2009

